### ISSUE TITLE
pyuv-1.4.0 not supported for interpreter python3.10

### ISSUE THREAD

**Issue by ijaketak** [NixOS/nixpkgs#179857]
### Steps To Reproduce
1. build `powerline`

### Build log
```
error: pyuv-1.4.0 not supported for interpreter python3.10
(use '--show-trace' to show detailed location information)
```

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.197, NixOS, 22.11 (Raccoon), 22.11pre385637.3d7435c638b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(root): `"home-manager, musnix, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Artturin** [comment]
https://github.com/saghul/pyuv/issues/273

use `python39Packages.powerline` and if you can please submit a pr changing

```
  powerline = with python3Packages; toPythonApplication powerline;
```

to

```
  # pyuv doesn't build on python3.10 https://github.com/saghul/pyuv/issues/273
  powerline = with python39Packages; toPythonApplication powerline;
```

**Comment by bjornfor** [comment]
Upstream has merged the [python-3.10 build fix](https://github.com/saghul/pyuv/pull/275). Perhaps we can backport that until a new release comes out?
